### PR TITLE
[ticket/14596] Deny installation of Ascraeus under PHP 7

### PR DESCRIPTION
**Pull request by 3D-I**
https://tracker.phpbb.com/browse/PHPBB3-14596

PHPBB3-14596

### CODE REVIEW

**Comment by bantu** [comment]
Your seem to be assuming the installer actually works on PHP7. Are you sure that is the case?

**Comment by marc1706** [comment]
The installer works fine on PHP7. The biggest issues is with custom BBCodes using the /e modifier.

**Comment by 3D-I** [comment]
I checked the installer also live on my webspace under PHP 7.0.4, before of this proposal.

File: phpBB/language/en/install.php [hunk 1]
```diff
@@ -300,11 +300,11 @@
 	'PHP_REGISTER_GLOBALS'			=> 'PHP setting <var>register_globals</var> is disabled',
 	'PHP_REGISTER_GLOBALS_EXPLAIN'	=> 'phpBB will still run if this setting is enabled, but if possible, it is recommended that register_globals is disabled on your PHP install for security reasons.',
 	'PHP_SAFE_MODE'					=> 'Safe mode',
-	'PHP_SETTINGS'					=> 'PHP version and settings',
-	'PHP_SETTINGS_EXPLAIN'			=> '<strong>Required</strong> - You must be running at least version 5.3.3 of PHP in order to install phpBB. If <var>safe mode</var> is displayed below your PHP installation is running in that mode. This will impose limitations on remote administration and similar features.',
+	'PHP_SETTINGS'					=> 'PHP versions and settings',
```
Comment (lavigor) [inline phpBB/language/en/install.php]: Why this change to `versions`?
The board actually works under a single PHP version.
The same for the line below.

**Comment by 3D-I** [comment]
Versions means (for me) the range of PHP versions in between phpBB 3.1.x works.
I could be wrong - being myself not a native speaker - tough.

**Comment by lavigor** [comment]
@3D-I here the word 'version' is not related to the requirement. It is related to the board that is installed on a certain version of PHP.
In other words `PHP version >= 5.3.3, < 7.0.0-dev` means `Is your board's PHP version higher or equal to 5.3.3 and less than 7.0.0-dev?`. Yes/No answer is displayed in the next column.

Using your edit `PHP versions: >= 5.3.3, < 7.0.0-dev` the question is `Are all of your board's PHP versions higher or equal to 5.3.3 and less than 7.0.0-dev?`. Do you actually use different PHP versions for different board directories? Probably no.

BTW I think this is not a language-related issue but a semantic one.

**Comment by 3D-I** [comment]
I will be more than glad to change it if I have to. 
Sorry but I do not have so much time for to discuss about minutia, thanks. :)

**Comment by JimMH** [comment]
Any updates on perhaps merging this? It is causing confusion with community members.

https://www.phpbb.com/community/viewtopic.php?p=14469171#p14469171

**Comment by lavigor** [comment]
@JimMH this PR still has minor issues, but I think that it will be ready before 3.1.10. 😉

**Comment by 3D-I** [comment]
Which are those minor issues?

**Comment by lavigor** [comment]
@3D-I I pointed them out in my comments above.
I think that any issue needs fixing if it was found.

File: phpBB/language/en/install.php (hunk 1 above)
Comment (VSEphpbb) [inline phpBB/language/en/install.php]: agreed

**Comment by 3D-I** [comment]
So this line should be written like

**PHP version and settings**

or what else?

**Comment by marc1706** [comment]
Yes, correct. We'll probably have to prepare another PR for the merge conflict this will cause with 3.2.x. If you're not too experienced in how we handle separate PRs for different branches, we can of course take care of that for you once this PR is merge-ready (should be once the setting is renamed as mentioned).

**Comment by 3D-I** [comment]
Please confirm I have to change the two lines: **303** and **307** into the language file.
I am ready to push the new commit but I do prefer to wait for a response prior to do that.

@marc1706  And yes, once done I would prefer you guys to take care of the 3.2.x thingh since I do not know from where to start from for that. Else I will be more than glad to learn if somebody gives to me some hint/docs.

But better to make this story short, I guess. ;)

Regards.

**Comment by lavigor** [comment]
@3D-I please don't be afraid of making commits.
You can rebase/create new commits/create new PRs if something is not right.
I can do the commit for you if you want so that you can cherry-pick/merge it to your branch.

Result:
**line 303**:
```
	'PHP_SETTINGS'					=> 'PHP version and settings',
```
**line 307**:
```
	'PHP_VERSION_REQD'				=> 'PHP version >= 5.3.3, < 7.0.0-dev',
```

P.S. I would have made other changes too, but these two are enough for this PR to be OK.

**Comment by 3D-I** [comment]
Thanks, this first PR against phpBB gave a lot of stress to me and others, I know.
Those changes I have already made prior to post here today and I am ready to push that commit, as I said.

**Comment by lavigor** [comment]
@3D-I oh, no! ... 😭
Please do not create a new PR when you receive an error from Travis.
I can explain you how to rebase the current branch in Git Bash if you want.

**Comment by 3D-I** [comment]
Ahah, you know the reason why I didn't try to commit any more things here.

Yes, please. I am using tortoise-git but I do have also GIT bash and I know it.
Better to switch to PMs or IRC official channel?

**Comment by lavigor** [comment]
@3D-I PM or Skype probably.

**Comment by 3D-I** [comment]
I am not using skypee anymore, sorry. I am now online on .com or area51.
Else the dev-channel of phpBB? Just tell me.

TIA.

**Comment by Crizz0** [comment]
Short description:
Use gitbash in your repository-dir:
git checkout PHPBB3-14596 (switch to your branch)

git rebase -i HEAD~1 (this will rebase the last commit)

Then you'll see something like:

> pick  de76b6b https://tracker.phpbb.com/browse/PHPBB3-14596

Now you've to change the "pick" to "reword" (be careful with "insert/delete"-command on your keyboard)

Change the commit-message to:

> [ticket/14596] Changes versions to version 
> 
> PHPBB3-14596

Then press ESC --> type in ":wq" -> press enter.

The rebase will be made and correct your commit message. After wards push the changes again to github. Maybe a force push: 
`git push -f <remotename> <branchname>`
 is needed. :) 

If you have trouble, contact us in #phpbb-dev on freenode: irc://chat.freenode.net/phpbb-dev

**Comment by VSEphpbb** [comment]
Lose the semi-colon and fix the commit message

**Comment by 3D-I** [comment]
Thanks @lavigor @Crizz0  ..
I got a missing DLL in my installation of GIT for Window$ for some unknown reason, I am fixing my dev-environment and I will commit again.

Your patience is much appreciated.

**Comment by 3D-I** [comment]
Damn AVG Pro that messed up my GIT Bash, got to reinstall it.
Followed the instruction by @lavigor via PM.

About the 3.2.x's possible conflict If somebody is willing to tell me what to do I willdo, instead feel free you guys to take care of it.

**Comment by 3D-I** [comment]
Should I do some thing more here guys?

**Comment by 3D-I** [comment]
Thanks.
